aptos-move/framework/aptos-token/sources/property_map.move

```diff
@@ -160,6 +162,32 @@ module aptos_token::property_map {
         *property_val = value;
     }
 
+    public fun create_property_value_raw(
+        value: vector<u8>,
+        type: String
+    ): PropertyValue {
+        PropertyValue {
+            value,
+            type,
+        }
+    }
+
+    /// create a property value from generic type data
+    public fun create_property_value<T: copy>(data: &T): PropertyValue {
+        if (
+            type_name<T>() == string::utf8(b"bool") ||
+            type_name<T>() == string::utf8(b"u8") ||
+            type_name<T>() == string::utf8(b"u64") ||
+            type_name<T>() == string::utf8(b"u128") ||
+            type_name<T>() == string::utf8(b"address") ||
+            type_name<T>() == string::utf8(b"0x1::string::String")
+        ) {
+            create_property_value_raw(bcs::to_bytes<T>(data), type_name<T>())
```

Comment:
nit, you're likely going to perform the same function each time, I doubt move compiler has any sort of optimizations at this point in time.

recommend:

```
let name = type_name<T>();
if (name == ... || name == ...) { create_property_value_raw(..., name))